### DISCUSSION TITLE
In-line citation links only link years??

### DISCUSSION THREAD

**Discussion by noahmf** [quarto-dev/quarto-cli#8834] [Q&A]
### Description

I am using the code below.

The paren cite is fully hyperlinked.  But for the in-line citation, **only the year is hyperlinked**.

I have tried biblatex and custom .csl files to no avail.  natbib gets the hyperlinking right, but it doesn't have the citation style I want to use (Chicago author-date).

Is there a way to get around this?

Thank you!

````qmd
---
title: "Paper 123"
date: today
format: 
  pdf:
    documentclass: article
    link-citations: true
bibliography: references.bib
---

## Introduction

...

Test citation here [@tan2023] and here @tan2023.

\newpage

## References {.unnumbered}

\setstretch{1}

::: {#refs}
:::
````

**Comment by mcanouil** [comment]
I am almost certain that it is a LaTeX thing that you can investigate using `keep-tex: true`, then looking at the LaTeX code.

PS: you should not use bare LaTeX code.
- See [markdown basics](https://quarto.org/docs/authoring/markdown-basics.html#page-breaks) to learn how to make a page break.
- Use `` `\setstretch{1}`{=latex} ``  not `\setstretch{1}`.

**Reply by noahmf** [reply]
Thanks.  I have looked at the latex code using `keep-tex: true`.

Unfortunately, Quarto outputs to .tex like this:

`Test citation here (\citeproc{ref-tan2023}{Tan 2023}) and here Tan
(\citeproc{ref-tan2023}{2023}).`

So I don't think there's any (straightforward) way to e.g., fix the problem by renewing a latex command in an "include-in-header"?  Because for the in-line cite, "Tan" is outside the `\citeproc` call.

**Reply by mcanouil** [reply]
I am not sure that's Quarto as the citation mechanism used is the one from Pandoc.

You can try using `quarto pandoc index.qmd --from markdown --to latex -o index.tex` and see the output of Pandoc itself.

**Reply by noahmf** [reply]
Unfortunately I know very little about Pandoc.

But it looks like this:

`Test citation here {[}@tan2023{]} and here @tan2023.`

**Reply by mcanouil** [reply]
Sorry, I forgot a flag to Pandoc:

```sh
quarto pandoc index.qmd --from markdown --to latex -o index.tex --citeproc
```

<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/7f0bbe8a-a934-4477-bb41-660cb1de66ef">

So as I assumed, this is all done by Pandoc.
You can change the `cite-method` to not use `citeproc`, see <https://quarto.org/docs/output-formats/pdf-basics.html#citations>.
For instance `natbib`:

<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/6fee7a0b-4152-485f-a211-4abda90779f5">

> I have tried biblatex and custom .csl files to no avail. natbib gets the hyperlinking right, but it doesn't have the citation style I want to use (Chicago author-date).

Where did you looked exactly?

If you follow the links from <https://quarto.org/docs/authoring/footnotes-and-citations.html#sec-citations-style>, for example Zotero and search for Chicago: <https://www.zotero.org/styles?q=chicago>.
I can see "[Chicago Manual of Style 16th edition (author-date)](https://www.zotero.org/styles/chicago-author-date-16th-edition)"

**Reply by cderv** [reply]
> So as I assumed, this is all done by Pandoc.

Yes this is Pandoc citation processing which is writing the tex output like this 
```tex
Test citation here (\citeproc{ref-tan2023}{Tan 2023}) and here Tan (\citeproc{ref-tan2023}{2023}).
```
when `link-citations` is opt-in. `\citeproc` command they defined, is only applied on the year. 

Maybe there would be a specific "workaround" with a Lua filter to catch the [Cite](https://pandoc.org/lua-filters.html#type-cite) node and output raw latex directly so that you do this before pandoc does it. but I did not try and not sure it is possible. 

Anyhow, if you think this is not expected, you should report to Pandoc directly. 

> I have tried biblatex and custom .csl files to no avail. natbib gets the hyperlinking right, but it doesn't have the citation style I want to use (Chicago author-date).

To be clear, CSL only works with Pandoc citeproc method. For natbib or biblatex, you need to use a `biblio-style` (documented at [Quarto doc](https://quarto.org/docs/output-formats/pdf-basics.html#citations) and [Pandoc doc](https://pandoc.org/MANUAL.html#biblatex-bibliographies))

So you need a BST file for the Chicago style.
